rcl/include/rcl/event.h

```diff
@@ -50,13 +50,15 @@ typedef enum rcl_subscription_event_type_t
 typedef struct rmw_event_t rmw_event_t;
 
 /// Internal rcl implementation struct.
-struct rcl_event_impl_t;
+struct rcl_event_impl_s;
+
+typedef struct rcl_event_impl_s rcl_event_impl_t;
```

Comment:
Do we actually need this `typedef`?  We only ever use `struct rcl_event_impl_s` in this header file, and the typedef to `rcl_event_impl_t` is done in event_impl.h, so I *think* we can just drop this.

Reply:
The thing is that we do use some of these `_impl_t` types downstream (e.g. `rcl_service_impl_t` in `rcl_action`). I chose not to break API for them and to be consistent i.e. to have a `typedef` for all.

It does seem I can get rid of the `struct` forward declaration if the `typedef` is in place. See https://github.com/ros2/rcl/pull/932/commits/beda62e00bd1e8c320ff67171146c11df61b2c27.